app/src/main/java/com/example/admin/personallibrarycatalogue/AddBookActivityFragment.java

```diff
@@ -0,0 +1,191 @@
+                    Toast.makeText(getActivity(), PLEASE_ENTER_THE_TITLE_FIELD, Toast.LENGTH_SHORT).show();
+                    return;
+                }
+
+                Book book = new Book(id_,title, author, description, cover);
+
+                // in case user edit information just update data
+                if (id_ != null) {
+                    helper.updateBook(id_, book);
+                } else {
+                    helper.addBook(book);
+                }
+
+                Intent intent = new Intent(getActivity(), BooksListActivity.class);
+                startActivity(intent);
+
+            }
+        });
+
+        // Actions for cancel button (clear fields and leave activity)
+        Button cancelButton = (Button) rootView.findViewById(R.id.cancel);
+        cancelButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                ViewGroup group = (ViewGroup) rootView.findViewById(R.id.add_book_layout);
+                clearForm(group);
+
+                Intent intent = new Intent(getActivity(), MainActivity.class);
+                startActivity(intent);
+            }
+        });
+
+        return rootView;
+    }
+
+    /**
+     * Clears all edtiText fileds
+     */
+    private void clearForm(ViewGroup group) {
+        for (int i = 0, count = group.getChildCount(); i < count; i++) {
+            View view = group.getChildAt(i);
+            if (view instanceof EditText) {
+                ((EditText) view).setText("");
+            }
+        }
+    }
+
+    private void fillAllFields(View view, LibraryDatabaseHelper helper, int id) {
+
+        Book book = helper.getBookById(id);
+
+        EditText titleEditText = (EditText) view.findViewById(R.id.title_edit_text);
+        titleEditText.setText(book.getTitle());
+
+        EditText authorEditText = (EditText) view.findViewById(R.id.author_edit_text);
+        authorEditText.setText(book.getAuthor());
+
+        EditText descriptionEditText = (EditText) view.findViewById(R.id.description_edit_text);
+        descriptionEditText.setText(book.getDescription());
+
+        ImageView coverView = (ImageView) view.findViewById(R.id.cover_image_view);
+        coverView.setImageBitmap(Util.getBitmapFromBytes(book.getCover()));
+
+    }
+
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent imageReturnedIntent) {
+        super.onActivityResult(requestCode, resultCode, imageReturnedIntent);
+        switch (requestCode) {
+            case SELECT_PHOTO:
+                if (resultCode == getActivity().RESULT_OK) {
+                    try {
+                        final Uri imageUri = imageReturnedIntent.getData();
+                        final InputStream imageStream = getActivity().getContentResolver().openInputStream(imageUri);
+                        final Bitmap selectedImage = BitmapFactory.decodeStream(imageStream);
```

Comment:
OutOfMemoryError is here. 
1. Start adding a book
2. Change cover
3. Change cover again and again until crash happens.